Python/libraries/recognizers-sequence/recognizers_sequence/sequence/extractors.py

```diff
@@ -114,3 +115,19 @@ def __init__(self):
             ReVal(RegExpUtility.get_safe_reg_exp(BasePhoneNumbers.SpecialPhoneNumberRegex),
                   Constants.PHONE_NUMBER_REGEX_SPECIAL),
         ]
+
+class BaseEmailExtractor(SequenceExtractor):
+    @property
+    def _extract_type(self) -> str:
+        return 'email'
+
+    @property
+    def regexes(self) -> List[ReVal]:
+        return self._regexes
+
+    def __init__(self):
+        self._regexes = [
+            ReVal(RegExpUtility.get_safe_reg_exp(BaseEmail.EmailRegex), Constants.EMAIL_REGEX),
+            #EmailRegex2 will break the code as it's not supported in Python, comment out for now
```

Comment:
What is not supported in it?

Reply:
The EmailPrefix & EmailSuffix in base_emial.py have a little bug for Regex or Re in Python do not have (?(condition) YES|No) but should use (?(?=condition) YES|No). Here is the code which I tested and passed.
`EmailPrefix = f'(?(?=\"\")(\"\".+?(?<!\\\\)\"\")|(([0-9A-Za-z]((\\.(?!\\.))|[-!#\\$%&\'\\*\\+/=\\?\\^\\{{\\}}\\|~\\w])*)(?<=[0-9A-Za-z])))'`
`EmailSuffix = f'(?(?=\\[)(\\[{IPv4Regex}\\])|{NormalSuffixRegex})'`
But it seems like to change the Base-Email.yaml in Patterns to fixed this bug. C#/Java/Python both have the "?=" and JS are not supported so I think it will work but my disk is not enough for building the entire project. Sorry for it. : (

Reply:
Got it. Thanks.